src/templates_front/templator_front.org

```diff
@@ -5775,7 +5812,174 @@ trexio_safe_to_orbital_list_up_dn (const int32_t N_int,
 }
     #+end_src
 
-     #+begin_src c :tangle trexio_private.h
+    #+begin_src c :tangle prefix_front.c
+trexio_exit_code
+trexio_convert_nao_radius_32 (const float r, const float* const grid_r, float* const log_r_out)
+{
+  if (r < 0) return TREXIO_INVALID_ARG_1;
+  if (grid_r == NULL) return TREXIO_INVALID_ARG_2;
+
+  *log_r_out = log(r / grid_r[0]) / log(grid_r[1] / grid_r[0]);
+
+  return TREXIO_SUCCESS;
+}
+
+trexio_exit_code
+trexio_convert_nao_radius_64 (const double r, const double* const grid_r, double* const log_r_out)
+{
+  if (r < 0) return TREXIO_INVALID_ARG_1;
+  if (grid_r == NULL) return TREXIO_INVALID_ARG_2;
+
+  *log_r_out = log(r / grid_r[0]) / log(grid_r[1] / grid_r[0]);
+
+  return TREXIO_SUCCESS;
+}
+
+trexio_exit_code
+trexio_convert_nao_radius_py (const double r, double* grid_r, int32_t n_grid, double* const log_r_out)
+{
+  if (r < 0) return TREXIO_INVALID_ARG_1;
+  if (grid_r == NULL) return TREXIO_INVALID_ARG_2;
+
+  *log_r_out = log(r / grid_r[0]) / log(grid_r[1] / grid_r[0]);
+
+  return TREXIO_SUCCESS;
+}
+
+  #+end_src
+
+  #+begin_src c :tangle prefix_front.c
+trexio_exit_code
+trexio_evaluate_nao_radial (const int32_t shell_index, const double r, const int32_t* const grid_start,
+                            const int32_t* const grid_size, const double* const grid_r, const double* const interpolator, const double* const normalization, double* const amplitude)
+{
+  if (shell_index < 0) return TREXIO_INVALID_ARG_1;
+  if (r < 0) return TREXIO_INVALID_ARG_2;
+  if (grid_start == 0) return TREXIO_INVALID_ARG_3;
+  if (grid_size == 0) return TREXIO_INVALID_ARG_4;
+  if (grid_r == NULL) return TREXIO_INVALID_ARG_5;
+  if (interpolator == 0) return TREXIO_INVALID_ARG_6;
+  if (normalization == 0) return TREXIO_INVALID_ARG_7;
+
+  const int32_t i0 = 4*grid_start[shell_index];
+
+  // Convert radius to logarithmic units 
+  double r_log = 0;
+  trexio_convert_nao_radius_64 (r, grid_r + grid_start[shell_index], &r_log);
+  int32_t i_log = (int32_t) r_log;
+  if (i_log < 0)
+    i_log = 0;
+  else if (i_log >= grid_size[shell_index])
+    return 0; // NAOs vanish at the boundary by definition
```

Comment:
We return `trexio_exit_code` not integer values.

Reply:
@scemama Could you have a closer look here?

Reply:
I agree. Something is strange here. I believe you should set `*amplitude = 0.` and return `TREXIO_SUCCESS`.